gluten-substrait/src/main/scala/org/apache/spark/sql/execution/ApplyResourceProfileExec.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution
+
+import org.apache.gluten.execution.GlutenPlan
+import org.apache.gluten.extension.columnar.transition.Convention
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.resource.ResourceProfile
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, SortOrder}
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, OrderPreservingUnaryNode}
+import org.apache.spark.sql.catalyst.plans.physical.{Distribution, Partitioning}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+private case class ApplyResourceProfileExecAdaptor(child: LogicalPlan)
+  extends OrderPreservingUnaryNode {
+  override def output: Seq[Attribute] = child.output
+
+  // For spark 3.2.
+  protected def withNewChildInternal(newChild: LogicalPlan): ApplyResourceProfileExecAdaptor =
+    copy(child = newChild)
+}
+
+/**
+ * Used to apply specified resource profile for the whole stage.
+ * @param child
+ * @param resourceProfile
+ *   resource profile specified for child belong stage.
+ */
+case class ApplyResourceProfileExec(child: SparkPlan, resourceProfile: ResourceProfile)
+  extends UnaryExecNode
+  with GlutenPlan {
+
+  if (child.logicalLink.isDefined) {
+    setLogicalLink(ApplyResourceProfileExecAdaptor(child.logicalLink.get))
+  }
```

Comment:
Missed a question here. Is there specific reason of setting the logical link with an `ApplyResourceProfileExecAdaptor`?

Reply:
Actually no. Just keep consistent with `FakeRowLogicAdaptor`.

Reply:
> Actually no. Just keep consistent with FakeRowLogicAdaptor.

I see. that one may have specific reason, I am not sure.

So can we remove `ApplyResourceProfileExecAdaptor`? I feel `ApplyResourceProfileExecAdaptor`'s logical link should be the same with its child's logical link. Since it and its child seem to be logically equivalent (share the same schema, output, statistics, etc.). Thanks.

Reply:
addressed.